test/unit/proxy_statuses/proxy_status_puppetca_test.rb

```diff
@@ -0,0 +1,31 @@
+require 'test_helper'
+
+class ProxyStatusPuppetcaTest < ActiveSupport::TestCase
+  setup do
+    @proxy = FactoryGirl.build_stubbed(:smart_proxy, :url => 'https://secure.proxy:4568')
+    certificates = {"secure.proxy"=>{"state"=>"valid", "fingerprint"=>"SHA256", "serial"=>3, "not_before"=>"2015-12-12T14:33:10UTC", "not_after"=>"2020-12-11T14:33:10UTC"}, "proxy.host"=>{"state"=>"valid", "fingerprint"=>"SHA256", "serial"=>3, "not_before"=>"2015-12-22T14:33:10UTC", "not_after"=>"2020-12-22T14:33:10UTC"}}
+    autosign = ['autosigned.com']
+    ProxyAPI::Puppetca.any_instance.stubs(:all).returns(certificates)
+    ProxyAPI::Puppetca.any_instance.stubs(:autosign).returns(autosign)
+    @proxy_status = ProxyStatus::PuppetCA.new(@proxy)
+  end
+
+  test 'it returns CA certificates' do
+    cert = @proxy_status.certs[0]
+    assert_kind_of SmartProxies::PuppetCACertificate, cert
+    assert_equal(cert.name, @proxy.hostname)
+    assert_equal(cert.state, "valid")
+    assert_equal(cert.fingerprint, "SHA256")
+    assert_equal(cert.valid_from, Time.parse("2015-12-12T14:33:10UTC"))
+    assert_equal(cert.expires_at, Time.parse("2020-12-11T14:33:10UTC"))
+    assert_equal(cert.smart_proxy_id, @proxy.id)
+  end
+
+  test 'it returns expiry for oldest certificate' do
+    assert_equal(Time.parse("2020-12-11T14:33:10UTC"), @proxy_status.expiry)
+  end
+
+  test 'it returns autosign' do
+    assert_equal('autosigned.com', @proxy_status.autosign[0])
```

Comment:
Is this testing anything at all? autosign is stubbed above, maybe you want to `expect` a proxy call or something similar